core/contractHost.js

```diff
@@ -46,79 +52,103 @@ No invites left`;
     ).then(_ => seats.get(seatIdentity));
   }
 
-  // The contract host is designed to have a long-lived credible
-  // identity.
+  function evaluateStringToFn(contractSrcString) {
+    insist(typeof contractSrcString === 'string')`\n
+"${contractSrcString}" must be a string, but was ${typeof contractSrcString}`;
+    const fn = evaluate(contractSrcString, {
+      Nat,
+      harden,
+      console,
+      E,
+      makePromise,
+      // TODO: sameStructure is used in one check...() function. Figure out a
+      // more general approach to providing useful helpers.
+      sameStructure,
+    });
+    insist(typeof fn === 'function')`\n
+"${contractSrcString}" must be a string for a function, but produced ${typeof contractSrcString}`;
+    return fn;
+  }
+
+  /**
+   * Build an object containing functions with names starting with 'check' from
+   * strings in the input contract.
+   */
+  function extractCheckFunctions(contractSrcs) {
```

Comment:
This seems a little brittle to me.

Reply:
Agreed. I suggested this particular hack so that we could get the rest working. Now that it is, we should figure out what we want instead. If we come up with something small and easy it can go in the PR. But it would be fine with me if we instead filed an issue on this hack now and fixed it in a subsequent PR.

Reply:
Sounds good to me!